microservices.yml

```diff
@@ -156,7 +159,16 @@ jobs:
       - print-diagnostics
       - fetch-environment-variables-from-s3
 
-  # Step 2: Verify that tests are passing
```

Comment:
Why would we want to push an image with potentially failing tests?

Reply:
I see no purpose pushing builds with failing tests to ECR. Maybe just save to workspace?

Reply:
i;m all on not pushing failing builds. 
but how do you think we should approach this; as each "step" is run on different executor so built image needs to be "transferred" to next executor, if we don't use some repo...

any ideas?

Reply:
`setup_remote_docker` step is not compatible with machine executor... 

> Build-agent version 0.1.1250-22bf9f5d (2018-12-12T11:32:15+0000)
> Configuration errors: 1 error occurred:
> 
> * In step 6 (setup_remote_docker): Step isn't compatible with machine executor.  You can use Docker without a special step.

so `persist` and `attach`?

Reply:
`persist && attach` "example": https://circleci.com/workflow-run/c818a3d5-743f-4f7c-a898-aed7fb5d15ef

Reply:
sounds good to leverage the workspace and docker for this.  do remember that the `machine` executor is required for our vpn access only, so we cannot remove it or use `setup_remote_docker`.

Reply:
I've updated this PR in the way that now the "push to ECR" step can require the tests to pass

an example of this working is in https://github.com/MGMResorts/devices/pull/416